rust_src/src/frame.rs

```diff
@@ -857,96 +857,103 @@ pub extern "C" fn do_switch_frame(
     // This used to check for a live frame, but apparently it's possible for
     // a switch-frame event to arrive after a frame is no longer live,
     // especially when deleting the initial frame during startup.
-    if let Some(mut target_frame) = frame.as_live_frame() {
-        if target_frame.eq(&current_frame_ref) {
```

Comment:
I think this one got lost in the translation to a match statement. It could be added back my adding a
```rust
    Some(target_frame) if target_frame.eq(&current_frame_ref) => return frame,
```
branch to the start of the match